### PR TITLE
Add annotation related metadata to the API

### PR DESCRIPTION
**Pull request by jotare**
### Description
Related with DONUT training.
This is the first part, with API and storing, dataset/train in another PR

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb_models/nucliadb_models/metadata.py [hunk 1]
```diff
@@ -292,6 +307,15 @@ def from_message(cls: Type[_T], message: resources_pb2.UserFieldMetadata) -> _T:
             including_default_value_fields=True,
             use_integers_for_enums=True,
         )
+        value["selections"] = [
+            MessageToDict(
+                selections,
+                preserving_proto_field_name=True,
+                including_default_value_fields=True,
+                use_integers_for_enums=True,
```
Comment (lferran) [inline nucliadb_models/nucliadb_models/metadata.py:315]: don't we want to serialize the enums with the string instead of with the integer?

**Review by lferran — APPROVED** [review]
lgtm! 👍🏼 

this doesn't need any extra logic in the ingest so that the data is persisted because it's inside the basic protobuf, right?

File: nucliadb_models/nucliadb_models/metadata.py (hunk 1 above)
Comment (jotare) [reply nucliadb_models/nucliadb_models/metadata.py:315]: Not sure about this, I'm using the same `MessageToDict` as above

**Comment by jotare** [comment]
> lgtm! 👍🏼
> 
> this doesn't need any extra logic in the ingest so that the data is persisted because it's inside the basic protobuf, right?

I don't think so. As metadata is already stored as a protobuffer inside the basic

**Pull request merged**